app/controllers/register/establishers/company/CompanyAddressListController.scala

```diff
@@ -36,43 +38,51 @@ import viewmodels.address.AddressListViewModel
 import scala.concurrent.{ExecutionContext, Future}
 
 class CompanyAddressListController @Inject()(
-                                              override val appConfig: FrontendAppConfig,
-                                              val userAnswersService: UserAnswersService,
-                                              @EstablishersCompany override val navigator: Navigator,
-                                              override val messagesApi: MessagesApi,
-                                              authenticate: AuthAction,
-                                              getData: DataRetrievalAction,
-                                              allowAccess: AllowAccessActionProvider,
-                                              requireData: DataRequiredAction
-                                            )(implicit val ec: ExecutionContext) extends AddressListController with Retrievals {
+    override val appConfig: FrontendAppConfig,
+    val userAnswersService: UserAnswersService,
+    @EstablishersCompany override val navigator: Navigator,
+    override val messagesApi: MessagesApi,
+    authenticate: AuthAction,
+    getData: DataRetrievalAction,
+    allowAccess: AllowAccessActionProvider,
+    requireData: DataRequiredAction,
+    val auditService: AuditService
+)(implicit val ec: ExecutionContext)
+    extends AddressListController
+    with Retrievals {
 
   def onPageLoad(mode: Mode, srn: Option[String], index: Index): Action[AnyContent] =
     (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async { implicit request =>
-      viewmodel(mode, srn, index).right.map(get)
+      (CompanyDetailsId(index) and CompanyPostCodeLookupId(index)).retrieve.right
```

Comment:
This logic can be moved back to the viewmodel instead of repeating it in get and post methods